### PR TITLE
examples/dmodex.c: Fix some bugs.

### PR DESCRIPTION
**Pull request by awlauria**
- Fence after the loop to prevent a hang if one rank fails.
- Fix check for a local rank.
- Make sure the namespace and rank is always copied in proc.

Signed-off-by: Austen Lauria <awlauria@us.ibm.com>

### CODE REVIEW

**Comment by awlauria** [comment]
This test passes for me on a single node and 2 nodes.

**Comment by awlauria** [comment]
bot:retest

**Comment by rhc54** [comment]
> This test passes for me on a single node and 2 nodes.

Well...yes, though you have eliminated a significant element of the test - namely, the entire "wait until data is published" portion of the code. Thus, you won't see any issues with the hotel eviction/recheck code path, etc.

In other words, this "fixed" the bug by simply ignoring the faulty code path 😄

**Comment by awlauria** [comment]
I see what you mean regarding the first fence  - I thought it was required to distribute the data before the Get() calls. Will remove it.

The second fence (after the loop) should be after the `done:` label, or else ranks on a failure will go straight to PMIx_Finalize(), while those who succeeded will be stuck in the fence. I think this one should stay as is.

File: examples/dmodex.c [hunk 1]
```diff
@@ -227,16 +228,16 @@ int main(int argc, char **argv)
         }
     }
 
+done:
+    /* finalize us */
+
     /* call fence so everyone waits before leaving */
-    proc.rank = PMIX_RANK_WILDCARD;
-    if (PMIX_SUCCESS != (rc = PMIx_Fence(&proc, 1, NULL, 0))) {
+    if (PMIX_SUCCESS != (rc = PMIx_Fence(NULL, 0, NULL, 0))) {
```
Comment (rhc54) [inline examples/dmodex.c:235]: Just an FYI (so no need to change anything): `PMIx_Fence` with a `NULL` proc argument defaults to "all procs in my nspace".

**Pull request merged**